### PR TITLE
[#160616] Adds ActiveStorage feature flag

### PR DESCRIPTION
**Pull request by jossim**
# Release Notes

- [x] Add specs for `read_attached_file`
- [x] Refactor the skipped `journal_spec`
- [x] Add to the readme
- [x] Consider refactoring `DownloadableFile` into 2 modules, `DownloadableFile` and `DownloadablePaperclipFile`

### CODE REVIEW

File: config/secrets.yml.template [hunk 1]
```diff
@@ -17,6 +17,13 @@ defaults: &defaults
   #   aws_access_key_id: <%= ENV.fetch("AWS_ACCESS_KEY", "Your-Key-Here") %>
   #   aws_secret_access_key: <ENV.fetch("AWS_SECRET_ACCESS_KEY", "Your-Key-Here") %>
 
+  # To use ActiveStorage, add your credentials here and enable Azure or AWS in your environment file: config.active_storage.service = :azure
+  # active_storage:
+  #   azure:
+  #     storage_account_name: <%= ENV.fetch("AZURE_STORAGE_ACCOUNT", "Your-Key-Here") %>,
+  #     storage_access_key: <%= ENV.fetch("AZURE_STORAGE_ACCESS_KEY", "Your-Key-Here") %>,
```
Comment (giladshanan) [inline config/secrets.yml.template:24]:
```suggestion
  #     storage_account_name: <%= ENV.fetch("AZURE_STORAGE_ACCOUNT", "Your-Key-Here") %>
  #     storage_access_key: <%= ENV.fetch("AZURE_STORAGE_ACCESS_KEY", "Your-Key-Here") %>
```

File: app/models/concerns/active_storage_file.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    end
+  end
+
+  def file_path
+    # service is nil when no file is attached
+    file.service&.path_for("file")
+  end
+
+  def rename(attr, value)
+    file.upate(attr, value)
```
Comment (jmyers0022) [inline app/models/concerns/active_storage_file.rb:50]: Hope you don't mind me poking my head in here. Should this be `update`?

**Comment by jossim** [comment]
@giladshanan looks good!

File: app/models/concerns/active_storage_file.rb (hunk 2 above)
Comment (giladshanan) [reply app/models/concerns/active_storage_file.rb:50]: @jmyers0022 not at all, good catch!  much appreciated

Comment (giladshanan) [inline app/models/concerns/active_storage_file.rb:50]:
```suggestion
    file.update(attr, value)
```
There are a bunch of specs that would catch this if the feature flag is set to `active_store: true`.

**Review by giladshanan — APPROVED** [review]
Approving on behalf of @jossim

**Pull request merged**